osu.Game/Screens/Play/HUD/GameplaySpectatorList.cs

```diff
@@ -0,0 +1,148 @@
+                    Child = new Container
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.TopLeft,
+                        Origin = Anchor.TopLeft,
+                        Child = SpectatorText = new OsuSpriteText
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            Anchor = Anchor.TopLeft,
+                            Origin = Anchor.TopLeft,
+                            Colour = Color4.White,
+                            Font = OsuFont.Torus.With(size: 14, weight: FontWeight.SemiBold),
+                            Truncate = false,
+                            Shadow = false,
+                            Margin = new MarginPadding { Left = GameplaySpectatorUser.SHEAR_WIDTH },
+                        },
+                    },
+                },
+                Flow = new FillFlowContainer<GameplaySpectatorUser>
+                {
+                    RelativeSizeAxes = Axes.X,
+                    X = GameplaySpectatorUser.SHEAR_WIDTH,
+                    Anchor = Anchor.TopLeft,
+                    Origin = Anchor.TopLeft,
+                    AutoSizeAxes = Axes.Y,
+                    Direction = FillDirection.Vertical,
+                    Spacing = new Vector2(2.5f),
+                    LayoutDuration = 450,
+                    LayoutEasing = Easing.OutQuint,
+                    // Make the margin a bit higher than the text height
+                    Margin = new MarginPadding { Top = OsuFont.Torus.Size * 1.5f },
+                },
+            };
+
+            // Hide by default
+            Flow.Alpha = 0.0f;
+            OtherSpectators.Alpha = 0.0f;
+
+            // When the bindable changes, update the display
+            MaxNames.ValueChanged += _ => updateDisplay();
+            configVisibility.ValueChanged += _ => updateDisplay();
+        }
+
+        public GameplaySpectatorUser Add(IUser? user)
+        {
+            var drawable = CreateSpectatorUserDrawable(user);
+
+            Flow.Add(drawable);
+
+            updateDisplay();
+            return drawable;
+        }
+
+        public void Remove(IUser? user)
+        {
+            Flow.RemoveAll(u => u.User?.Username == user?.Username, true);
+
+            updateDisplay();
+        }
+
+        private void updateDisplay()
+        {
+            // We removed everything, fade out everything
+            if (Spectators == 0 || configVisibility.Value == false)
+            {
+                Flow.FadeOut(fade_time);
+                OtherSpectators.FadeOut(fade_time);
+                return;
+            }
+
+            // Fade in the Text if needed
+            if (OtherSpectators.Alpha == 0.0f)
+            {
+                OtherSpectators.FadeIn(fade_time);
+            }
+
+            // always adjust text
+            SpectatorText.Text = GameplaySettingsStrings.Spectators + " (" + Spectators + ")";
```

Comment:
Concatenating `LocalisableString`s and plain strings defeats the point of the `LocalisableString`. This is doing an impliciit conversion from `LocalisableString` to `string` (to perform the concatenation - this loses the localisable property of the string), and then back from `string` to `LocalisableString`. You want `LocalisableString.Format()` instead.

Reply:
Thanks, was unaware 😔